theano/gpuarray/opt.py

```diff
@@ -358,9 +357,10 @@ def apply(self, fgraph):
                             [mapping[i] for i in node.inputs],
                             node.outputs)
                     finally:
-                        time_opts[lopt] += time.time() - t_opt
-                        self.new_opts.append(lopt)
+                        t_opt2 = time.time()
                     if new_ops:
+                        self.new_opts.append(lopt)
+                        time_opts[lopt] = max(time_opts[lopt], (t_opt2 - t_opt))
```

Comment:
I think the old code was better. You should not do that in the if, but outside as we need to compute the time even if the opt didn't apply. Also, we need to add the time for each lopt, not take the max. The problem wasn't there revert to the old code.